ios/browser/api/networking/brave_certificate_model.mm

```diff
@@ -0,0 +1,473 @@
+    
+    //Certificate Transparency Extensions
+    #ifndef OPENSSL_NO_CT
+    {NID_ct_precert_scts, [BraveCertificateSCTExtensionModel class]},
+    {NID_ct_cert_scts, [BraveCertificateSCTExtensionModel class]}
+    #elif NID_ct_precert_scts && NID_ct_cert_scts // Chromium's BoringSSL is missing so much stuff!
+    {NID_ct_precert_scts, [BraveCertificateGenericExtensionModel class]},
+    {NID_ct_cert_scts, [BraveCertificateGenericExtensionModel class]}
+    #endif
+  };
+  
+  NSMutableArray* parsed_extensions = [[NSMutableArray alloc] init];
+  const STACK_OF(X509_EXTENSION)* extensions_list = X509_get0_extensions(x509_cert_);
+  if (extensions_list) {
+    std::size_t count = sk_X509_EXTENSION_num(extensions_list); //X509_get_ext_count
+    
+    for (std::size_t i = 0; i < count; ++i) {
+      X509_EXTENSION* extension = sk_X509_EXTENSION_value(extensions_list, static_cast<int>(i)); //X509_get_ext
+      if (extension) {
+        ASN1_OBJECT* object = X509_EXTENSION_get_object(extension);
+        int nid = OBJ_obj2nid(object);
+        auto supported_extension = supported_extensions.find(nid);
+        if (supported_extension != supported_extensions.end() && supported_extension->second != nullptr) {
+          // All supported extensions MUST be a BraveCertificateExtensionModel
+          // Otherwise they should be parsed as a BraveCertificateGenericExtensionModel or using their own model
+          if (![supported_extension->second isSubclassOfClass:[BraveCertificateExtensionModel class]]) {
+            NSAssert(NO, @"INVALID EXTENSION CLASS!");
+            continue;
+          }
+          
+          // Extension is supported
+          BraveCertificateExtensionModel* model = [[supported_extension->second alloc] initWithType:supported_extension->first withExtension:extension];
+          [parsed_extensions addObject:model];
+        } else {
+          // Unsupported extension
+          BraveCertificateExtensionModel* model = [[BraveCertificateGenericExtensionModel alloc] initWithType:nid withExtension:extension];
+          [parsed_extensions addObject:model];
+        }
+      }
+    }
+  }
+  
+  _extensions = parsed_extensions;
+}
+
+- (void)debugExtensions {
+  const STACK_OF(X509_EXTENSION)* extensions_list = X509_get0_extensions(x509_cert_);
+  if (extensions_list) {
+    std::size_t count = sk_X509_EXTENSION_num(extensions_list); //X509_get_ext_count
+    
+    for (std::size_t i = 0; i < count; ++i) {
+      X509_EXTENSION* extension = sk_X509_EXTENSION_value(extensions_list, static_cast<int>(i)); //X509_get_ext
+      if (extension) {
+        ASN1_OBJECT* object = X509_EXTENSION_get_object(extension);
+        bool is_critical = X509_EXTENSION_get_critical(extension);
+        
+        std::string result = x509_utils::string_from_ASN1_OBJECT(object, false);
+        result += ": ";
+        result += is_critical ? "true" : "false";
+        result += "\n--------------------------\n";
```

Comment:
More of a heads up in case you hadn't worked with them before- when there's a bunch of std::string concatenations like this, sometimes it's handy to use `std::stringstream` (or `std::ostringstream`; both part of <sstream>). Should be usable in Objective C 😄 No speed benefit that I know of though, just (potentially) cleaner formatting

ex:
```
std::ostringstream result(x509_utils::string_from_ASN1_OBJECT(object, false));
result << ": " << is_critical << endl;
result << "--------------------------" << endl;

// ...

std::cout << endl << result.str() << endl;
```